dm/dm/worker/server.go

```diff
@@ -58,11 +58,19 @@ var (
 // dispatches requests to worker
 // sends responses to RPC client.
 type Server struct {
-	sync.Mutex
+	// closeMu is used to protect 5 fields below
+	closeMu    sync.Mutex
+	closed     atomic.Bool
+	inited     bool
```

Comment:
`inited` doesn't reflect its meaning accurately which is whether we have called start/close ever.

if we change  call sequence in main from 
```
go signal-process-routine which calls s.Close
s.Start
s.Close
```
to this, we can avoid the chance of calling Close before Start
```
s.Init
go signal-process-routine which calls s.Close
s.Serve
s.Close
```
or better we can run s.Serve in a separate routine

but it's ok for now.